src/OpenSage.Mathematics/IRandom.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+
+namespace OpenSage.Mathematics;
+
+public interface IRandom
+{
+    uint Seed { get; }
+
+    void Initialize(uint seed);
+
+    int Next(int lo, int hi);
+    float NextSingle(float lo, float hi);
+}
+
+public sealed class SystemRandom : IRandom
+{
+    private Random _random;
+
+    public uint Seed => throw new NotImplementedException();
+
+    public SystemRandom()
+    {
+        _random = CreateRandom(0);
+    }
+
+    public void Initialize(uint seed)
+    {
+        _random = CreateRandom((int)seed);
+    }
+
+    private static Random CreateRandom(int seed) => new(seed);
+
+    public int Next(int lo, int hi)
+    {
+        return _random.Next(lo, hi + 1);
+    }
```

Comment:
I think this is counterintuitive for most developers experienced with csharp. If I have a Random.Next method, I'm expecting the high value to be exclusive, not inclusive.

I know this is how the existing implementation functioned, so I think if we do something like this we should name the method something different (e.g. `NextInclusive` or similar) so it's clear what's happening here (I know that would result in many more lines changed).

Just my $0.02

Reply:
That does make sense. On the other hand, I'd prefer for such a commonly-used method to have a short name. To be consistent, we'd have `NextSingleInclusive` or `NextLogicFrameSpanInclusive`, and that's arguably less readable. I'm not sure it's worth the benefit of clarity for developers used to `System.Random`. But I could be persuaded if you think we should.

Reply:
Yeah, makes sense. The name was just an example, but I don't know that I have a better one in mind.

If nothing else, I might at least suggest some clear documentation on the interface?

Reply:
> If nothing else, I might at least suggest some clear documentation on the interface?

Good idea - I've [done that](https://github.com/OpenSAGE/OpenSAGE/pull/1326/commits/7f79f0dd2c60749c3805f44025fb2f191aaff623).